### PR TITLE
Scan for changes in upstream licenses

### PR DESCRIPTION
**Pull request by andrewmwells-amazon**
## Description of changes

Scan for changes in upstream licenses.

## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [x] A change "invisible" to users (e.g., documentation, changes to "internal" crates like `cedar-policy-core`, `cedar-validator`, etc.)

I confirm that this PR (choose one, and delete the other options):

- [x] Does not update the CHANGELOG because my change does not significantly impact released code.

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Does not require updates because my change does not impact the Cedar Dafny model or DRT infrastructure.

## Disclaimer

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by aaronjeline — APPROVED** [review]

**Comment by john-h-kastner-aws** [comment]
Where's the check actually being done?

**Comment by andrewmwells-amazon** [comment]
> Where's the check actually being done?

Once we merge this we'll add it from https://github.com/marketplace/actions/cargo-deny

**Comment by andrewmwells-amazon** [comment]
> > Where's the check actually being done?
> 
> Once we merge this we'll add it from https://github.com/marketplace/actions/cargo-deny

NVM, I assumed the big green button meant I wouldn't have to edit `yml` files. I'll just do it manually.

**Review by aaronjeline — APPROVED** [review]

**Review by john-h-kastner-aws — APPROVED** [review]

**Review by shaobo-he-aws — APPROVED** [review]

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -27,7 +27,8 @@ jobs:
       - run: cargo audit --deny warnings # For some reason this hangs if you don't cargo build first
 
   cargo-deny:
+    name: cargo deny
     runs-on: ubuntu-22.04
     steps:
-    - uses: actions/checkout@v3
-    - uses: EmbarkStudios/cargo-deny-action@v1
+      - uses: actions/checkout@v3
+      - run: EmbarkStudios/cargo-deny-action@v1
```
Comment (andrewmwells-amazon) [inline .github/workflows/build_and_test.yml:34]:
```suggestion
      - uses: EmbarkStudios/cargo-deny-action@v1
```

**Review by shaobo-he-aws — APPROVED** [review]

**Pull request merged**